lib/ws_servers/api/handlers/on_algo_order_remove.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const sendError = require('../../../util/ws/send_error')
+const validateParams = require('../../../util/ws/validate_params')
+const isAuthorized = require('../../../util/ws/is_authorized')
+
+module.exports = async (server, ws, msg) => {
+  const { algoDB, as } = server
+
+  const [, authToken, exID, orders] = msg
+
+  const validRequest = validateParams(ws, {
+    exID: { type: 'string', v: exID },
+    authToken: { type: 'string', v: authToken },
+    orders: { type: 'array', v: orders }
+  })
+
+  if (!validRequest) {
+    return
+  }
+
+  if (!isAuthorized(ws, authToken)) {
+    return sendError(ws, 'Unauthorized')
+  } else if (exID !== 'bitfinex') {
+    return sendError(ws, 'Algo orders currently only enabled for Bitfinex')
+  } else if (!ws.aoc) {
+    return sendError(ws, 'Unauthorized')
+  }
+
+  for (const algOrder of orders) {
+    const { gid, algoID } = algOrder
+    const { AlgoOrder } = algoDB
+    await AlgoOrder.update({ gid, algoID }, { active: false })
+  }
+
+  as.open()
```

Comment:
currently `as.open()` is called in these cases:

 - no algo order found
 - algo orders were removed
 - dms = false

am i missing something or is there a case missing were someone does not want to remove any orders and the aos is not started? afaik the current implementation requires a request to "remove algo orders" with an empty array in those cases?

Reply:
how about extending the PR a bit:

1. the algo order host is always started 
2. db is read from the outside, then algo orders that are supposed to start are sent to the algo host
3. the remove endpoint just removes algo orders from the db
4. on start of the algo server, no algo orders are started automatically (i.e. you don't pass active algo orders into the constructor)